bin/addons-linter

```diff
@@ -7,23 +7,39 @@ global.appRoot = path.relative(process.cwd(), absoluteAppRoot);
 global.localesRoot = path.join(absoluteAppRoot, 'dist', 'locale');
 global.nodeRequire = require;
 
-try {
-  // `../dist/addons-linter` doesn't exist if the linter has not been
-  // built yet. Disable this eslinting so that users don't get confused
-  // needlessly and aren't required to build the linter for eslint
-  // to pass.
-  // eslint-disable-next-line import/no-unresolved
-  require('../dist/addons-linter').createInstance({ runAsBinary: true }).run();
-} catch (err) {
-  if (err.code === 'MODULE_NOT_FOUND') {
-    console.log('You did not build addons-linter yet.');
-    console.log(
-      'Please run `npm install` and `npm run build` or see README.md for more information.'
-    );
+(async () => {
+  let instance;
 
-    process.exitCode = 1;
-    return;
-  }
+  try {
+    // `../dist/addons-linter` doesn't exist if the linter has not been
+    // built yet. Disable this eslinting so that users don't get confused
+    // needlessly and aren't required to build the linter for eslint
+    // to pass.
+    // eslint-disable-next-line import/no-unresolved
+    instance = require('../dist/addons-linter').createInstance({
+      runAsBinary: true,
+    });
+    await instance.run();
+  } catch (err) {
+    if (err.code === 'MODULE_NOT_FOUND') {
+      console.log('You did not build addons-linter yet.');
+      console.log(
+        'Please run `npm install` and `npm run build` or see README.md for more information.'
+      );
+
+      process.exitCode = 1;
+      return;
+    }
 
-  throw err;
-}
+    // Print the error message without the stack if it is a user error
+    // and --stack cli option wasn't part of the cli option passed to the
+    // linter.
+    if (err.name === 'AddonsLinterUserError' && !instance.config.stack) {
+      console.error(err.message);
+      process.exitCode = 1;
```

Comment:
Maybe add a different exit code here like 2 or 127.

Reply:
Also, question: why do we use `process.exitCode` and not `process.exit(1)`?

Reply:
> Maybe add a different exit code here like 2 or 127.

sure sounds good to me too, In 661f1cf I did also add one small integration test while I was here ;-) 

> Also, question: why do we use `process.exitCode` and not `process.exit(1)`?

I was honestly asking myself the same thing, mostly out of curiosity.

My guess was that `process.exitCode + return` may easier to unit test (because the nodejs process would  be forced to exit and so in theory a unit test could require that module and then check that process.exitCode has been set to the expected value)... but I looked back in the history (because I wasn't really working on this repo at the time) but I didn't find a reasoning for preferring this particular form (the change has been introduced in #1492) and the change didn't land with a unit test (which doesn't exclude that maybe it was meant to be a follow up, but I wasn't able to track down if that was the case).

Well, unfortunately this brief historical research didn't got me any additional detail about the reasons behind that :-P, but in the end I don't mind that we use `process.exitCode + return` instead of `process.exit` :-D